public/app/app.ts

```diff
@@ -256,6 +258,19 @@ export class GrafanaApp {
 
       setReturnToPreviousHook(useReturnToPreviousInternal);
 
+      if (config.featureToggles.preserveDashboardStateWhenNavigating) {
```

Comment:
although this has just made me think of something...

in both cases (either with this logic or using `window.sessionStorage`), i don't think you'll then be able to open a dashboard in a new tab and have it preserve the variables. is that a use case we're considering? 🤔

Reply:
> i don't think you'll then be able to open a dashboard in a new tab and have it preserve the variables. is that a use case we're considering?

I think this is an improvement I'm going to consider after we land this PR. It will require some dynamic generation of the url, maybe based on a specific use interaction (i.e. key + click)